### PR TITLE
[tests] fix currently failing unittests

### PR DESCRIPTION
**Pull request by rainisto**

### CODE REVIEW

**Comment by pgerdt** [comment]
Fixes most of the tests, but the test ut_recentcontactsmodel RecentContactsModelTest::simple() fails quite often, although it does pass occasionally. Here's the output:

FAIL!  : RecentContactsModelTest::simple() Compared lists have different sizes.
   Actual   (e.contacts()) size: '2'
   Expected (QList<ContactDetails>() << qMakePair(aliceId, aliceName)) size: '1'
   Loc: [recentcontactsmodeltest.cpp(179)]

**Comment by rainisto** [comment]
Thanks, I'll check if that can be fixed too...

**Comment by rainisto** [comment]
I was not able to reproduce with:
for ((i=1; i <= 10; ++i)); do ./ut_recentcontactsmodel; done 2>&1 | grep Totals

Are you sure that you didn't have multiple instances running / runned tests are nemo? Or should I do try 100 times?

File: tests/ut_eventmodel/eventmodeltest.cpp [hunk 1]
```diff
@@ -716,6 +716,8 @@ void EventModelTest::testDeleteEventGroupUpdated()
 
     QVERIFY(groupModel.databaseIO().getGroup(group.id(), group));
 
+    QTest::qWait(1000); // wait for database to update correctly
```
Comment (amtep) [inline tests/ut_eventmodel/eventmodeltest.cpp]: Are you sure there's no signal or you can wait for? Or a condition using QTRY_COMPARE?

File: tests/ut_callmodel/callmodeltest.cpp [hunk 2]
```diff
@@ -1013,12 +1013,13 @@ void CallModelTest::testModifyEvent()
 // minimize to the same number.
 void CallModelTest::testMinimizedPhone()
 {
+    QSKIP("Contact matching is not yet supported with SQLite");
     deleteAll();
 
     const QString phone00("0011112222");
-    const QString phone99("0011112222");
+    const QString phone99("9911112222");
```
Comment (amtep) [inline tests/ut_callmodel/callmodeltest.cpp]: Thanks for catching this :)

**Comment by amtep** [comment]
@pgerdt is it possible that you have some junk in your contacts db that matches alicePhone1? Its number is 1234567 which is something you may have used in your own testing.
(Which is a reason to change it in the test I think)

File: tests/ut_eventmodel/eventmodeltest.cpp (hunk 1 above)
Comment (rainisto) [inline tests/ut_eventmodel/eventmodeltest.cpp]: yes, QTRY_COMPARE is cleaner solution, updating pull request....

**Comment by pgerdt** [comment]
@amtep: I did not have a contact with the id 1234567 in my contacts db.

@rainisto: The recentcontactsmodel  tests started working after I nuked my contacts db, and added  a single new contact.

**Comment by rainisto** [comment]
Good to hear that test is not failing anymore. If lgtm then feel free to merge (I don't have write access to this repo).

**Pull request merged**